core/server/master/src/test/java/alluxio/master/AlluxioMasterProcessTest.java

```diff
@@ -179,6 +192,58 @@ public void startMastersThrowsUnavailableException() throws InterruptedException
     Assert.assertTrue(success.get());
   }
 
+  @Test
+  public void failToGainPrimacyWhenJournalCorrupted() throws Exception {
+    Configuration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.UFS);
+    URI journalLocation = JournalUtils.getJournalLocation();
+    JournalSystem journalSystem = new JournalSystem.Builder()
+        .setLocation(journalLocation).build(CommonUtils.ProcessType.MASTER);
+    AlluxioMasterProcess masterProcess = new AlluxioMasterProcess(journalSystem);
+    corruptJournalAndStartMasterProcess(masterProcess, journalLocation);
+  }
+
+  @Test
+  public void failToGainPrimacyWhenJournalCorruptedHA() throws Exception {
+    Configuration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.UFS);
+    URI journalLocation = JournalUtils.getJournalLocation();
+    JournalSystem journalSystem = new JournalSystem.Builder()
+        .setLocation(journalLocation).build(CommonUtils.ProcessType.MASTER);
+    ControllablePrimarySelector primarySelector = new ControllablePrimarySelector();
+    FaultTolerantAlluxioMasterProcess masterProcess =
+        new FaultTolerantAlluxioMasterProcess(journalSystem, primarySelector);
+    primarySelector.setState(PrimarySelector.State.PRIMARY);
+    corruptJournalAndStartMasterProcess(masterProcess, journalLocation);
+  }
+
+  private void corruptJournalAndStartMasterProcess(AlluxioMasterProcess masterProcess,
+      URI journalLocation) throws Exception {
+    assertTrue(masterProcess.mJournalSystem instanceof UfsJournalSystem);
+    masterProcess.mJournalSystem.format();
+    // corrupt the journal
+    UfsJournal fsMaster =
```

Comment:
Is it possible to use the actual UfsJournal -> gainPrimacy -> write() -> write?
I am asking to avoid the use of Mockito and can test the more close to real case